packages/http-client/src/ceramic-http-client.ts

```diff
@@ -223,11 +263,7 @@ export class CeramicClient implements CeramicApi {
     const type = stream.state.type
     const streamConstructor = this._streamConstructors[type]
     if (!streamConstructor) throw new Error(`Failed to find constructor for stream ${type}`)
-    return new streamConstructor(stream, this.context) as T
-  }
-
-  async setDID(did: DID): Promise<void> {
-    this.context.did = did
+    return new streamConstructor(stream, this) as T
```

Comment:
I do not quite like passing full `this` around the code.

Reply:
same as above. can split it out in this PR, it just becomes a lot bigger.